### ISSUE TITLE
Integrate Hub into Web Editor

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-web-editor#25]
When selecting a server it would be great if you could get a list of potential back-ends from the Hub! 

Maybe also add a feature to search for an alternative back-end based on the loaded process graph.

**Comment by m-mohr** [comment]
@christophfriedrich Is there a public API I could query against?

**Comment by christophfriedrich** [comment]
Yes, the Hub exposes a public API. The Hub's frontend itself queries its data from that API!

There is no documentation of that API (yet), but the endpoints are defined in the Hub's [`server.js`](https://github.com/Open-EO/openeo-hub/blob/master/server.js) file if you want to have a look (search for `server.get` and `server.post`).

In short, if you just want a list of the backends: GET http://hub.openeo.org/backends :tipping_hand_man:

**Comment by christophfriedrich** [comment]
The Hub now "officially" exposes four public endpoints, see [its README](https://github.com/Open-EO/openeo-hub/blob/master/README.md#public-api-endpoints) for information/documentation.

**Comment by m-mohr** [comment]
Process graphs are now part of the Web Editor, server selection is still a todo.

**Comment by m-mohr** [comment]
The Hub will likely only be a back-end filter and not a sharing platform for processes or UDFs anytime soon. 
So the only place where I can add something is the "server selection" at the beginning (i.e. where we already have the list of servers stored locally).
I'm thinking about using vue-multiselect and replace the modal to select the server so that users can search for servers and pick them directly with the two data sources (1) locally and (2) Hub.

**Comment by christophfriedrich** [comment]
https://github.com/Open-EO/openeo-web-editor/pull/199#discussion_r701857220

> By the way, I'm planning to replace this [the server URL history list] with a vue-multiselect where you can search through previously connected servers and the Hub...

When this is done remember to adapt the help text too

**Comment by m-mohr** [comment]
> In short, if you just want a list of the backends: GET http://hub.openeo.org/backends 💁‍♂️

@christophfriedrich I guess this has been updated? I only get an error right now:

```
{
"code": "ResourceNotFound",
"message": "/backends; caused by Error: ENOENT: no such file or directory, stat 'dist/backends'"
}
```

**Comment by m-mohr** [comment]
Closing here as the future of the Hub is unclear and it doesn't make sense right now to investigate this.
